### ISSUE TITLE
(v2) Triggering intellisense in VS Code crashes the Tailwind extension when working on gluestack components

### ISSUE THREAD

**Issue by resticker** [gluestack/gluestack-ui#2382]
### Description

Showing intellisense options via the "Trigger Suggest" in the Command Palette (I use a keyboard shortcut like most of us probably do) causes the Tailwind extension to go to 100% CPU usage and become unresponsive

### CodeSandbox/Snack link

N/A

### Steps to reproduce

1. Go to `<directory with gluestack components>/button/index.tsx`
2. Put the cursor anywhere in the file
3. Select `Trigger Suggest` from the Command Palette
4. Notice that suggestions gets stuck on `Loading...` (see screenshot)
5. Open `Developer: Open Process Explorer` from the Command Palette and see `electron-nodejs (tailwindServer.js)` at 99 (% CPU)

Showing intellisense just gets stuck "Loading..."
![Screenshot 2024-08-13 at 8 50 59 PM](https://github.com/user-attachments/assets/645b39e2-ea8c-4dc8-a3b6-26dbe4ea08f3)

Process Explorer:
![tailwind usage](https://github.com/user-attachments/assets/a8b5e126-a100-4a67-b53a-3500d1bdbdf5)

### gluestack-ui Version

Latest

### Platform

- [ ] Expo
- [ ] React Native CLI
- [ ] Next
- [ ] Web
- [ ] Android
- [ ] iOS

### Additional Information

The extension needs to be manually restarted to get intellisense working again. No intellisense of any kind works in any file (even my personal files) while the extension is unresponsive

I'm using v0.12.6 of the official [Tailwind CSS IntelliSense](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss) extension.
Visual Studio Code Version: 1.92.1

I'm posting this issue here because it seems to be more of a gluestack issue than an extension issue.

**Comment by Viraj-10** [comment]
Hey @resticker,

Regarding the VS Code crashing issue, it seems to be related to the Tailwind plugin or a system issue. We don’t manipulate anything concerning VS Code extensions, and we have followed all the standards while creating a Tailwind config for gluestack.

- Please remove `gluestack-ui`, then try running the Tailwind plugin and check the CPU usage during the process.

**Comment by resticker** [comment]
@Viraj-10 I think the issue stems from the [gluestack tooling setup](https://gluestack.io/ui/docs/home/getting-started/tooling-setup) for intellisense.

If I remove this regex from my `settings.json` file. The error goes away.

Since this code is related to tailwind variants, I figured that there are likely issues over on that project. And, yep, here's a ["intellisense in tv config hangs" issue](https://github.com/nextui-org/tailwind-variants/issues/127)

If you scroll through those comments, you'll see that Tailwind Variants updated the regex ([pull request](https://github.com/nextui-org/tailwind-variants-docs/pull/34/files)), and it's been updated in the [Tailwind Variants docs](https://www.tailwind-variants.org/docs/getting-started#intellisense-setup-optional)

You can also comb through the Tailwind extension issue related to Tailwind Variants [here](https://github.com/tailwindlabs/tailwindcss-intellisense/issues/963#issuecomment-2126903740)

Basically, regex is the issue.

We might want to consider changing the [gluestack intellisense setup docs](https://gluestack.io/ui/docs/home/getting-started/tooling-setup) to use the [new Tailwind Variants regex](https://www.tailwind-variants.org/docs/getting-started#intellisense-setup-optional):

```["([\"'`][^\"'`]*.*?[\"'`])", "[\"'`]([^\"'`]*).*?[\"'`]"]```

instead of the current regex:

```["tva\\((([^()]*|\\([^()]*\\))*)\\)", "[\"'`]([^\"'`]*).*?[\"'`]"]```

I just tried it, and I'm getting intellisense in gluestack components, and no hanging. (A con to this, is that it looks like if you activate intellisense in any string, it offers tailwind class suggestions 😂.. but that's not all bad)

**Comment by luiz504** [comment]
> @Viraj-10 I think the issue stems from the [gluestack tooling setup](https://gluestack.io/ui/docs/home/getting-started/tooling-setup) for intellisense.
> 
> If I remove this regex from my `settings.json` file. The issue goes away.
> 
> Since this code is related to tailwind variants, I figured that there are likely issues over on that project. And, yep, here's a ["intellisense in tv config hangs" issue](https://github.com/nextui-org/tailwind-variants/issues/127)
> 
> If you scroll through those comments, you'll see that Tailwind Variants updated the regex ([pull request](https://github.com/nextui-org/tailwind-variants-docs/pull/34/files)), and it's been updated in the [Tailwind Variants docs](https://www.tailwind-variants.org/docs/getting-started#intellisense-setup-optional)
> 
> You can also comb through the Tailwind extension issue related to Tailwind Variants [here](https://github.com/tailwindlabs/tailwindcss-intellisense/issues/963#issuecomment-2126903740)
> 
> Basically, regex is the issue.
> 
> We might want to consider changing the [gluestack intellisense setup docs](https://gluestack.io/ui/docs/home/getting-started/tooling-setup) to match the [new Tailwind Variants regex](https://www.tailwind-variants.org/docs/getting-started#intellisense-setup-optional):
> 
> `` ["([\"'`][^\"'`]*.*?[\"'`])", "[\"'`]([^\"'`]*).*?[\"'`]"] ``
> 
> instead of the current regex:
> 
> `` ["tva\\((([^()]*|\\([^()]*\\))*)\\)", "[\"'`]([^\"'`]*).*?[\"'`]"] ``
> 
> I just tried it, and I'm getting intellisense in gluestack components, and no hanging. (A con to this, is that it looks like if you activate intellisense in any string, it offers tailwind class suggestions 😂.. but that's not all bad)

I was facing the same issue, found the same solution but, doing another fresh research on TV issues, I found this https://github.com/gluestack/gluestack-ui/issues/903#issuecomment-2341019720 
Doing some basic tests, its working without my CPU dying in the process  of saving the file :)
